### ISSUE TITLE
Improve documentation on client lifecycle & events

### ISSUE THREAD

**Issue by pedrolcn** [Unleash/unleash-client-node#181]
Based on the documentation alone there is not a clear distinction between the `ready` and `registered` events. The documentation describes them as follows:
 - `ready`: "is emitted once the fs-cache is ready. if no cache file exists it will still be emitted."
 - `registered`: "is emitted after the app has been registered at the api server".

It is not immediately clear which of these two events should I listen for if I want to know when the unleash client is ready to evaluate toggles. It also adds to the confusion that in the code example in the **Getting Started** section, the `ready` event is is annotated as an optional event and `registered` as a metrics event.

It would be nice to have clear documentation regarding the client lifecycle, specially to know after which event the client is ready to evaluate toggles.

As a benchmark the [node-redis](https://github.com/NodeRedis/node-redis#api) documentation could be use. I believe it does a good job in clarifying when exactly each event is emitted and why the library-user should care about them.

**Comment by ivarconr** [comment]
Thanks for pointing this out. I agree that this can be improved. In general all documentation for the client SDKs could benefit some cleaning up.
